src/tables.jl

```diff
@@ -207,16 +239,166 @@ function _colnames(s::AbstractDimStack)
 end
 
 
+# WideDimTable
+
+
+"""
+    WideDimTable <: AbstractDimTable
+
+    WideDimTable(A::AbstractDimArray)
+
+Construct a Tables.jl/TableTraits.jl compatible object out of an `AbstractDimArray`.
+
+This table will have a column for the array data and columns for each
+`Dimension` index, as a [`DimColumn`]. These are lazy, and generated
+as required.
+
+Column names are converted from the dimension types using
+[`DimensionalData.dim2key`](@ref). This means type `Ti` becomes the
+column name `:Ti`, and `Dim{:custom}` becomes `:custom`.
+
+To get dimension columns, you can index with `Dimension` (`X()`) or
+`Dimension` type (`X`) as well as the regular `Int` or `Symbol`.
+"""
+struct WideDimTable{DS} <: AbstractDimTable
+    colnames::Vector{Symbol}
+    dimcolumns::DS
+    dimarraycolumns::Vector{DimArrayColumn}
+end
+
+function WideDimTable(s::AbstractDimStack; mergedims=false)
+    dims_ = dims(s)
+    dimcolumns = collect(map(d -> DimColumn(d, dims_), dims_))
+    dimarraycolumns = collect(map(A -> DimArrayColumn(A, dims_), s))
+
+    if mergedims
+        dimcol = MergedDimColumn(Tuple(dimcolumns), :geometry)
```

Comment:
As mentioned in the issue, we can't have `:geometry` as a standard column in DimensionalData.

Thats the GeoInterface.jl spec, which is not so useful to the stats and machine learning people who also use this package. So we need to generalise this by passing in the name of the column and the dims that contribute to it to the `mergedims` keyword.

Then in Rasters.jl we can specify:

```julia
 mergedims=(XDim, YDim, ZDim) => :geometry
```

Rather than `true/false` values (default would be `nothing`). This would leave other dimensions lime `Ti` out of the `:geometry` column and give everyone a flexibility to control the colums of the table for various uses.

Reply:
I can make it so that callers can pick any name they want, but excluding some dimensions will create issues. For starters, we won't be able to dispatch on `MergedDimColumn` anymore, as we'll have to maintain a vector with a mixture of `MergedDimColumn` and `DimColumn`. We'll also need to consider that users can change the order of dimensions, which complicates recovering a `DimArray` from a table. Finally, figuring out how to recover a properly ordered set of dimensions from a mix of `MergedDimColumn` and `DimColumn` will be hard.

Reply:
After thinking a bit more, maybe instead of `MergedDimColumn`, we should just create a new merged dimension and then store it as a single `DimColumn`? The problem of recovering the original array from a table remains, however.

Reply:
I've added some code from #477 for merging dimensions, which seems to be working nicely. I think treating the new merged dimension as a single `DimColumn` is the way to go. The problem that remains is figuring out how to restore the original dimensions.

Reply:
Yeah thay makes sense. Half of my raster have a time dimension, having that merge with X/Y in the geometry tuple would be less than useful.

Reply:
`mergedims` now takes a dimension selector and allows the assignment of the merged dimension to a chosen name:

```julia
julia> dimarray = DimArray(rand(Float32, 512, 512, 5, 3), (X, Y, Dim{:band}, Dim{:time}));

julia> DimTable(dimarray) |> DataFrame
3932160×5 DataFrame
     Row │ X      Y      band   time   value    
         │ Int64  Int64  Int64  Int64  Float32  
─────────┼──────────────────────────────────────
       1 │     1      1      1      1  0.336031
       2 │     2      1      1      1  0.387499
       3 │     3      1      1      1  0.5118
       4 │     4      1      1      1  0.276879
    ⋮    │   ⋮      ⋮      ⋮      ⋮       ⋮
 3932157 │   509    512      5      3  0.991191
 3932158 │   510    512      5      3  0.126712
 3932159 │   511    512      5      3  0.991649
 3932160 │   512    512      5      3  0.565661
                            3932152 rows omitted

julia> DimTable(dimarray, layersfrom=Dim{:band}) |> DataFrame
786432×8 DataFrame
    Row │ X      Y      time   band_1    band_2      band_3      band_4    band_5    
        │ Int64  Int64  Int64  Float32   Float32     Float32     Float32   Float32   
────────┼────────────────────────────────────────────────────────────────────────────
      1 │     1      1      1  0.336031  0.220786    0.00182956  0.782216  0.0602413
      2 │     2      1      1  0.387499  0.00558186  0.419911    0.732554  0.396695
      3 │     3      1      1  0.5118    0.856497    0.257019    0.336152  0.585407
      4 │     4      1      1  0.276879  0.0037455   0.630669    0.952328  0.685334
   ⋮    │   ⋮      ⋮      ⋮       ⋮          ⋮           ⋮          ⋮          ⋮
 786429 │   509    512      3  0.617334  0.740707    0.152624    0.135545  0.991191
 786430 │   510    512      3  0.716968  0.865897    0.387702    0.654909  0.126712
 786431 │   511    512      3  0.683891  0.249271    0.946006    0.40959   0.991649
 786432 │   512    512      3  0.370734  0.222489    0.590784    0.803202  0.565661
                                                                  786424 rows omitted

julia> DimTable(dimarray, layersfrom=Dim{:band}, mergedims=(X,Y)=>:geometry) |> DataFrame
786432×7 DataFrame
    Row │ time   geometry    band_1    band_2      band_3      band_4     band_5    
        │ Int64  Tuple…      Float32   Float32     Float32     Float32    Float32   
────────┼───────────────────────────────────────────────────────────────────────────
      1 │     1  (1, 1)      0.336031  0.220786    0.00182956  0.782216   0.0602413
      2 │     2  (1, 1)      0.474556  0.425886    0.191104    0.0407686  0.402897
      3 │     3  (1, 1)      0.800677  0.319256    0.629539    0.987191   0.334838
      4 │     1  (2, 1)      0.387499  0.00558186  0.419911    0.732554   0.396695
   ⋮    │   ⋮        ⋮          ⋮          ⋮           ⋮           ⋮          ⋮
 786429 │     3  (511, 512)  0.683891  0.249271    0.946006    0.40959    0.991649
 786430 │     1  (512, 512)  0.603525  0.432016    0.899589    0.603063   0.437243
 786431 │     2  (512, 512)  0.41972   0.339989    0.863264    0.462888   0.360653
 786432 │     3  (512, 512)  0.370734  0.222489    0.590784    0.803202   0.565661
                                                                 786424 rows omitted
```